Src/Base/AMReX_MultiFabUtil.H

```diff
@@ -1160,13 +1153,92 @@ ReduceToPlane (int direction, Box const& domain, FabArray<FAB> const& mf, F cons
                 });
             }
 #endif
+}
+}
+
+template <typename Op, typename T, typename FAB, typename F,
+          std::enable_if_t<IsBaseFab<FAB>::value
+#ifndef AMREX_USE_CUDA
+                           && IsCallableR<T,F,int,int,int,int>::value
+#endif
+                           , int> FOO>
+BaseFab<T>
+ReduceToPlane (int direction, Box const& a_domain, FabArray<FAB> const& mf, F const& f)
+{
+    Box const domain = amrex::convert(a_domain, mf.ixType());
+
+    Box domain2d = domain;
+    domain2d.setRange(direction, 0);
+
+    T initval;
+    Op().init(initval);
+
+    BaseFab<T> r(domain2d);
+    r.template setVal<RunOn::Device>(initval);
+    auto const& ar = r.array();
+
+    for (MFIter mfi(mf,MFItInfo().UseDefaultStream().DisableDeviceSync());
+         mfi.isValid(); ++mfi)
+    {
+        Box bx = mfi.validbox() & domain;
+        if (bx.ok()) {
+            int box_no = mfi.LocalIndex();
+            detail::reduce_to_plane<Op, T>(ar, direction, bx, box_no, f);
         }
     }
     Gpu::streamSynchronize();
 
     return r;
 }
 
+template <typename FA, typename Op, typename T, typename FAB, typename F,
+          std::enable_if_t<IsBaseFab<FAB>::value
+                           && IsFabArray<FA>::value
+#ifndef AMREX_USE_CUDA
+                           && IsCallableR<T,F,int,int,int,int>::value
+#endif
+                           , int> FOO>
+std::pair<FA,FA>
+ReduceToPlaneMF (int direction, Box const& domain, FabArray<FAB> const& mf, F const& f)
+{
+    Box const ndomain = amrex::convert(domain, mf.ixType());
+
+    auto npts = amrex::convert(mf.boxArray(),IntVect(0)).numPts();
+    if (npts != ndomain.numPts()) {
+        amrex::Abort("ReduceToPlaneMF: mf's BoxArray must have a rectangular domain.");
+    }
+
+    Box domain2d = ndomain;
+    domain2d.setRange(direction, 0);
+
+    T initval;
+    Op().init(initval);
+
+    BoxList bl = mf.boxArray().boxList();
+    for (auto& b : bl) {
+        b.setRange(direction, 0);
+    }
+    BoxArray ba(std::move(bl));
+    FA tmpfa(ba, mf.DistributionMap(), 1, 0);
+
+    for (MFIter mfi(mf); mfi.isValid(); ++mfi)
+    {
+        Box bx = mfi.validbox() & ndomain;
+        if (bx.ok()) {
+            int box_no = mfi.LocalIndex();
+            detail::reduce_to_plane<Op, T>(tmpfa.array(mfi), direction, bx, box_no, f);
+        }
+    }
+
+    BoxArray ba2d(domain2d);
+    ba2d.maxSize(ba[0].length());
+    DistributionMapping dm2d(ba2d);
+    FA mf2d(ba2d, dm2d, 1, 0);
+    mf2d.ParallelAdd(tmpfa);
```

Comment:
This is only correct if it's reduce sum. We don't have a way for min and max yet. So maybe we can add a static assert there.